### PR TITLE
l3g4200d spi mode on imx6 iwave

### PR DESCRIPTION
**Pull request by alexkalmuk**
* Fix imx SPI (extract 2 buses, fix CS, fix init)
* Fix GPIO
* L3g4200d works on ecSPI2

### CODE REVIEW

File: src/drivers/gpio/gpio.c [hunk 1]
```diff
@@ -105,8 +105,8 @@ gpio_mask_t gpio_get(unsigned short port, gpio_mask_t pins) {
 	}
 	port = GPIO_PORT(port);
 	if (port >= chip->nports) {
-		log_error("port (%d) > chip->nports (%d)",
-			chip, port, chip->nports);
+		log_error("port (%d) is out of range (nports=%d)",
+			port, chip->nports - 1);
 		return -1;
```
Comment (anton-bondarev) [inline src/drivers/gpio/gpio.c]: return -EACCES;

File: src/drivers/gpio/imx/imx.c [hunk 2]
```diff
@@ -35,7 +33,7 @@ EMBOX_UNIT_INIT(imx_gpio_init);
 static struct gpio_chip imx_gpio_chip;
 
 static int imx_gpio_init(void) {
-	for (int i = 0; i <= GPIO_MAX; i++)
+	for (int i = 0; i <= GPIO_PORTS; i++)
```
Comment (anton-bondarev) [inline src/drivers/gpio/imx/imx.c]: for (int i = 0; i <= GPIO_PORTS; i++) {
}
Braces?

File: src/drivers/spi/imx6_ecspi/imx6_ecspi.h [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/**
+ * @file imx6_ecspi.c
```
Comment (anton-bondarev) [inline src/drivers/spi/imx6_ecspi/imx6_ecspi.h]: ```@file```

File: src/drivers/spi/imx6_ecspi/imx6_ecspi.c [hunk 4]
```diff
@@ -6,140 +6,146 @@
...
+	REG32_CLEAR(ECSPI_CONREG(dev), ECSPI_CONREG_CHANNEL_SELECT_MASK);
+	REG32_ORIN(ECSPI_CONREG(dev), dev->cs << ECSPI_CONREG_CHANNEL_SELECT_OFFT);
 
-	/* TODO config gpio in proper way for ECSPI2,3,4,5*/
-	gpio_n = imx6_ecspi_gpio_info[cs][0];
-	port   = imx6_ecspi_gpio_info[cs][1];
+	gpio_n = dev->cs_array[dev->cs][0];
+	port   = dev->cs_array[dev->cs][1];
 
-	gpio_setup_mode(gpio_n, 1 << port, GPIO_MODE_OUTPUT);
```
Comment (Deryugin) [inline src/drivers/spi/imx6_ecspi/imx6_ecspi.c]: Why remove this line?

File: src/drivers/spi/imx6_ecspi/imx6_ecspi.h (hunk 3 above)
Comment (alexkalmuk) [reply src/drivers/spi/imx6_ecspi/imx6_ecspi.h]: Fixed

File: src/drivers/gpio/gpio.c (hunk 1 above)
Comment (alexkalmuk) [reply src/drivers/gpio/gpio.c]: Thank you, fixed.

File: src/drivers/gpio/imx/imx.c (hunk 2 above)
Comment (alexkalmuk) [reply src/drivers/gpio/imx/imx.c]: Fixed.

**Pull request merged**